app/jobs/find_available_user_barcode_job.rb

```diff
@@ -57,20 +58,20 @@ def perform(user: nil, **args)
     # If we succeeded, it's time to tell the User object that its barcode is no longer pending
     if already_there == false
       begin
-        Rails.logger.debug "#{self.class.name}: barcode [#{user.barcode}] is available in Sierra, calling patron creator service."
+        Delayed::Worker.logger.info("#{self.class.name}: barcode [#{user.barcode}] is available in Sierra, calling patron creator service.")
 
         # TODO: on timeouts/exceptions/negative results,
         # don't send request to Patron Service, keep user as pending
-        user.send_request_to_patron_creator_service
+        user.send_request_to_patron_creator_service(pin_code)
```

Comment:
Why would `pin_code` here be different from `user.pin`?

Reply:
The user object isn't being serialized correctly, when passed to the ActiveJob, and I don't want to introduce the complexity of creating a custom serializer.

Reply:
Where is the serialization happening? Is this a known issue? How have ascertained if this being caused due to serialization?

Reply:
Objects sent to the ActiveJob perform process are serialized as part of the normal functionality.  This serialization is known to not always include all object fields.  One way to fix it is by writing a custom serializer for User, but I believe that would introduce unnecessary complexity.  If there was more than one field, or if there was some extra functionality I needed, I'd opt for the serializer, but in this case, I think passing the pin manually is the better choice.

Reply:
@daryanypl  https://blog.saeloun.com/2019/09/11/rails-6-custom-serializers-for-activejob-arguments.html

Reply:
There is also this, which mentions passing object IDs: https://jtway.co/its-time-for-active-job-239f7681a3ef

Reply:
As I've said, I do not believe writing a custom serializer is appropriate for this code.  This pull request is for a README file.  Let's have a meeting and talk about the serializer option at a later time.  I can make a backlog issue for it.